### PR TITLE
JSON definitions for colour blocks

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Replaces old colour block definitions with a Blockly.defineBlocksWithJsonArray(..) call. Generator unit tests continue to load and pass, signifying compatibility with prior block definitions.

Replaces extension 'math_number_tooltip' with the reusable 'parent_tooltip_when_inline' extension, also used by colour_picker. Includes tests.

### CODE REVIEW

File: blocks/colour.js [hunk 1]
```diff
@@ -20,6 +20,11 @@
 
 /**
  * @fileoverview Colour blocks for Blockly.
+ *
+ * This file is scrapped to extract .json file definitions. The array passed to
```
Comment (rachel-fenichel) [inline blocks/colour.js]: scraped, not scrapped.

File: blocks/colour.js [hunk 2]
```diff
@@ -20,6 +20,11 @@
...
+ * defineBlocksWithJsonArray(..) must be strict JSON: double quotes only, no
+ * outside references, no functions, etc. The one exception is end-of-line
+ * comments, which the scrapper will remove.
```
Comment (rachel-fenichel) [inline blocks/colour.js]: scraper, not scrapper

File: blocks/colour.js [hunk 3]
```diff
@@ -20,6 +20,11 @@
 
 /**
  * @fileoverview Colour blocks for Blockly.
+ *
+ * This file is scrapped to extract .json file definitions. The array passed to
+ * defineBlocksWithJsonArray(..) must be strict JSON: double quotes only, no
+ * outside references, no functions, etc. The one exception is end-of-line
```
Comment (rachel-fenichel) [inline blocks/colour.js]: no functions, no trailing commas, etc.

File: blocks/colour.js [hunk 4]
```diff
@@ -34,100 +39,92 @@ goog.require('Blockly.Blocks');
...
+        "align": "RIGHT"
+      }
+    ],
+    "output": "Colour",
+    "colour": "%{BKY_COLOUR_HUE}",
+    "helpUrl": "%{BKY_COLOUR_BLEND_HELPURL}",
+    "tooltip": "%{BKY_COLOUR_BLEND_TOOLTIP}"
   }
-};
+]);  // End defineBlocksWithJsonArray(...)
```
Comment (rachel-fenichel) [inline blocks/colour.js]: Is this line now required?  If so, add a comment and also use it in blocks/math.js

File: blocks/colour.js (hunk 1 above)
Comment (AnmAtAnm) [inline blocks/colour.js]: Fixed.

File: blocks/colour.js (hunk 3 above)
Comment (AnmAtAnm) [inline blocks/colour.js]: Added.

File: blocks/colour.js (hunk 2 above)
Comment (AnmAtAnm) [inline blocks/colour.js]: Fixed.

File: blocks/colour.js (hunk 4 above)
Comment (AnmAtAnm) [inline blocks/colour.js]: Added DO NOT DELETE. Will add the same when I get back to the math.js

**Pull request merged**